### ISSUE TITLE
Anomaly Detection in Video Sequence with Appearance-Motion Correspondence

### ISSUE THREAD

**Issue by llien30** [mn1204/paper_summary#38]
# INFO

## Author
Trong-Nguyen Nguyen, Jean Meunier
## Affiliation
DIRO, University of Montreal
## Conference or Year
ICCV 2019
## Link
- [arXiv](https://arxiv.org/abs/1908.06351)
- [official GitHub]()

# Abstract
Anomaly detection in surveillance video is difficult because it contains a many types of events

**Our Model = Reconstruction Network + Image translation model** 
- using same encoder for two subnetworks
- unsupervised learning
- estimate frame level score for unknown input

# Proposed Method
### Conv-AE + U-Net
<img width="794" alt="スクリーンショット 2020-05-05 22 32 58" src="https://user-images.githubusercontent.com/51707667/81071950-62f26680-8f20-11ea-9314-6353e7179044.png">

- This model does not use fully connected layer in anywhere
→ theoretically work on images of any resolution
- To keep the ratio of the surveillance video, the image size is set to a ratio of  1:1.5

### Discriminator
<img width="516" alt="スクリーンショット 2020-05-05 22 53 25" src="https://user-images.githubusercontent.com/51707667/81073887-3be96400-8f23-11ea-93a1-aca9793ab4e4.png">

### Loss Function
**Conv-AE**
→ **Detection of strange objects**
_Convolutional Layer + BatchNorm Layer + Leaky ReLU Layer_

<img width="249" alt="スクリーンショット 2020-05-05 22 50 10" src="https://user-images.githubusercontent.com/51707667/81073586-ced5ce80-8f22-11ea-8510-da07aed1a534.png">
<img width="316" alt="スクリーンショット 2020-05-05 22 50 38" src="https://user-images.githubusercontent.com/51707667/81073608-dac19080-8f22-11ea-8c37-158e600cbfbf.png">
<img width="304" alt="スクリーンショット 2020-05-05 22 50 53" src="https://user-images.githubusercontent.com/51707667/81073631-e3b26200-8f22-11ea-84a2-1c444415c4af.png">

### U-Net
→ **Detection of unusual motions**
GT of the Optical Flow is estimated by pretrained FlowNet2

Use L1 Norm ⇨ Reduce the effect of the things followed as below
- the result of FlowNet2 may contain noise (during the smoothing procedure)
- the GT can be obtained from other flownet → possibly have small patches of wrong and/or noisy motion measure
<img width="277" alt="スクリーンショット 2020-05-05 22 52 29" src="https://user-images.githubusercontent.com/51707667/81073783-1b210e80-8f23-11ea-8237-e10e108a0fa3.png">

### Objective Function
<img width="339" alt="スクリーンショット 2020-05-05 22 54 43" src="https://user-images.githubusercontent.com/51707667/81074033-6a673f00-8f23-11ea-9bec-2c98d0359d23.png">
<img width="349" alt="スクリーンショット 2020-05-05 22 54 59" src="https://user-images.githubusercontent.com/51707667/81074063-7521d400-8f23-11ea-849e-ada95f8140e4.png">

### Anomaly Score
<img width="305" alt="スクリーンショット 2020-05-05 22 57 20" src="https://user-images.githubusercontent.com/51707667/81074297-c8942200-8f23-11ea-8180-f593e1d17f44.png">
<img width="307" alt="スクリーンショット 2020-05-05 22 57 30" src="https://user-images.githubusercontent.com/51707667/81074321-cf229980-8f23-11ea-9976-48f8e3e9a2fb.png">
<img width="268" alt="スクリーンショット 2020-05-05 22 57 57" src="https://user-images.githubusercontent.com/51707667/81074374-df3a7900-8f23-11ea-9c5c-cee695aaebed.png">
<img width="285" alt="スクリーンショット 2020-05-05 22 58 09" src="https://user-images.githubusercontent.com/51707667/81074391-e5305a00-8f23-11ea-8177-ac3b664450eb.png">
<img width="239" alt="スクリーンショット 2020-05-05 22 58 19" src="https://user-images.githubusercontent.com/51707667/81074411-ec576800-8f23-11ea-97be-6b54895a9480.png">

# Evaluation
### Dataset
- CUHK Avenue
- UCSD Ped2
- Subway Entrance Gate and Exit Gate
- Traffic-Belleview and Traffic-Train

Training Data → Only normal events
measured by AUC of the ROC curve
<img width="1016" alt="スクリーンショット 2020-05-06 10 13 16" src="https://user-images.githubusercontent.com/51707667/81130291-3707cd00-8f82-11ea-927e-e0764207f81f.png">
<img width="737" alt="スクリーンショット 2020-05-05 23 01 02" src="https://user-images.githubusercontent.com/51707667/81074678-4d7f3b80-8f24-11ea-9a97-50bed747654b.png">
<img width="472" alt="スクリーンショット 2020-05-05 22 59 58" src="https://user-images.githubusercontent.com/51707667/81074571-26c10500-8f24-11ea-8a5e-8228985d695c.png">

# Contribution
- Combined Conv-AE and U-Net (trained end-to-end)
→ by sharing the same encoder, the model is forced to learn the correspondence
- proposed patch-based scheme estimating frame-level normality score
→ reduces the effect of noise
- competitive performance compared to SOTA methods(on 6 benchmark datasets)

# Discussion, Future Work

## Date
2020/05/05
